src/pip/_internal/commands/install.py

```diff
@@ -226,6 +228,31 @@ def add_options(self):
     @with_cleanup
     def run(self, options, args):
         # type: (Values, List[str]) -> int
+        # Warn if running pip with sudo
+        try:
+            if sys.platform == 'darwin' or sys.platform == 'linux':
+                is_sudo = os.getuid() == 0
+            elif sys.platform == 'win32' or sys.platform == 'cygwin':
+                is_sudo = ctypes.windll.shell32.IsUserAnAdmin() != 0
+        except AttributeError:
+            is_sudo = False
+
+        is_venv = (hasattr(sys, 'real_prefix')
+                      or (hasattr(sys, 'base_prefix')
+                      and sys.base_prefix != sys.prefix))
+
+        if is_sudo and not is_venv and not options.root_path:
+            command = path.basename(sys.argv[0])
+            if command == "__main__.py":
+                command = os.path.basename(sys.executable) + " -m pip"
+            sudo_warning = ("Running pip as root is not recommended! "
+                            "This could result in the corruption of the "
+                            "package directories when a package manager is "
+                            "used. Try `%s install --user` instead. "
+                            "See #6409 for the reason behind this warning."
```

Comment:
I am not sure all users will understand where to look for `#6409` or what it stands for.

Reply:
good point - removed